### PR TITLE
Google custom search v2

### PR DESCRIPTION
**Pull request by dbu**
TODO
* [ ] Bump composer to version 2.0
* [ ] Switch to PSR-2
* [ ] Cleanup DI configuration

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -39,7 +39,9 @@ public function getConfigTreeBuilder()
                     ->addDefaultsIfNotSet()
                     ->canBeUnset()
                     ->children()
-                        ->scalarNode('search_key')->defaultFalse()->end()
+                        ->scalarNode('search_key')->isRequired()->end()
+                        ->scalarNode('search_api_url')->defaultValue('https://www.googleapis.com/customsearch/v1')->end()
+                        ->scalarNode('api_key')->isRequired()->end()
```
Comment (dbu) [inline DependencyInjection/Configuration.php]: this must be set through the DI extension class

**Comment by ralf57** [comment]
Need to check if more item metadata need to be exposed

**Comment by dbu** [comment]
@ralf57 i can work on this tomorrow if you want. can you tell me if you see things that need to be done?

**Comment by lsmith77** [comment]
need a rebase now ..

**Comment by dbu** [comment]
superseeded by #17

**Pull request closed without merging**